api_app/db/repositories/resource_templates.py

```diff
@@ -97,6 +105,7 @@ def create_template(self, template_input: ResourceTemplateInCreate, resource_typ
             "resourceType": resource_type,
             "current": template_input.current,
             "required": template_input.json_schema["required"],
+            "authorizedRoles": template_input.json_schema["authorizedRoles"] if "authorizedRoles" in template_input.json_schema else [],
```

Comment:
I think we can't really make authorizedRoles a mandatory field at this point, as all customer templates will need to be migrated (not just the ones already in Cosmos, but any templates in their codebases too). Which is why we need to do these checks